src/go/rdctl/cmd/factoryReset.go

```diff
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
+	if err != nil {
+		return err
+	}
+	homeDir := os.Getenv("HOME")
+	rawPaths := []string{
+		".bashrc",
+		".bash_profile",
+		".bash_login",
+		".profile",
+		".zshrc",
+		".cshrc",
+		".tshrc",
+	}
+	for i, s := range rawPaths {
+		rawPaths[i] = path.Join(homeDir, s)
+	}
+	rawPaths = append(rawPaths, path.Join(configHomePath, "fish", "config.fish"))
+
+	return removePathManagement(rawPaths)
+}
+
+func deleteLimaVM() error {
+	err := setupLimaHome()
+	if err != nil {
+		return err
+	}
+	execPath, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	execPath, err = filepath.EvalSymlinks(execPath)
+	if err != nil {
+		return err
+	}
+	limactl := path.Join(path.Dir(path.Dir(execPath)), "lima", "bin", "limactl")
+	return exec.Command(limactl, "delete", "-f", "0").Run()
+}
+
+func removeDockerCliPlugins(altAppHomePath string) error {
+	cliPluginsDir := path.Join(dockerconfig.Dir(), "cli-plugins")
+	entries, err := os.ReadDir(cliPluginsDir)
+	if err != nil {
+		if errors.Is(err, syscall.ENOENT) {
+			// Nothing left to do here, since there is no cli-plugins dir
+			return nil
+		}
+		return err
+	}
+	for _, entry := range entries {
+		if entry.Type()&os.ModeSymlink != os.ModeSymlink {
+			continue
+		}
+		fullPathName := path.Join(cliPluginsDir, entry.Name())
+		target, err := os.Readlink(fullPathName)
+		if err != nil {
+			continue
+		}
+		if strings.Contains(target, altAppHomePath) {
+			os.Remove(fullPathName)
+		}
+	}
+	return nil
+}
+
+func removePathManagement(dotFiles []string) error {
+	for _, dotFile := range dotFiles {
+		byteContents, err := os.ReadFile(dotFile)
+		if err != nil {
+			if errors.Is(err, syscall.ENOENT) {
+				// Nothing left to do here, since there is no cli-plugins dir
+				continue
+			}
+			fmt.Fprintf(os.Stderr, "Error trying to read %s: %s\n", dotFile, err)
+			continue
+		}
+		contents := string(byteContents)
+		startTarget := "### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)"
+		endTarget := "### MANAGED BY RANCHER DESKTOP END (DO NOT EDIT)"
+		startPoint := strings.LastIndex(contents, startTarget)
+		if startPoint == -1 {
+			continue
+		}
+		relativeEndPoint := strings.Index(contents[startPoint:], endTarget)
+		if relativeEndPoint == -1 {
+			continue
+		}
+		newEndPoint := startPoint + relativeEndPoint + len(endTarget)
+		if len(contents) > newEndPoint && contents[newEndPoint] == '\n' {
+			newEndPoint += 1
+		}
+		newContents := contents[0:startPoint] + contents[newEndPoint:]
+		filestat, err := os.Stat(dotFile)
+		if err != nil {
+			return fmt.Errorf("error trying to stat %s: %w", dotFile, err)
+		}
+		err = os.WriteFile(dotFile, []byte(newContents), filestat.Mode())
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error trying to update %s: %s\n", dotFile, err)
+		}
+	}
+	return nil
+}
+
+type dockerConfigType map[string]interface{}
+
+func clearDockerContext() error {
+	// Ignore failure to delete this next file:
+	os.Remove(path.Join(dockerconfig.Dir(), "plaintext-credentials.config.json"))
```

Comment:
Don't we also need to delete the docker context (in `~/.docker/contexts/meta` or something, I believe)?

Reply:
This is managed by normal shutdown. But I added code in case RD hangs and the user had to kill the process and wants to ensure the CLI can remove everything.